src/Sylius/Bundle/CoreBundle/Controller/ProductController.php

```diff
@@ -114,6 +115,36 @@ public function indexByTaxonIdAction(Request $request, $id)
         return $this->renderResults($taxon, $paginator, 'productIndex.html', $request->get('page', 1));
     }
 
+    /**
+     * List products categorized under given archetype (fetch by its code or id).
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws NotFoundHttpException
+     */
+    public function indexByArchetypeAction(Request $request)
+    {
+        $archetypeIdOrCode = $request->attributes->get('archetype');
+        if (filter_var($archetypeIdOrCode, FILTER_VALIDATE_INT)) {
+            $archetype = $this->get('sylius.repository.product_archetype')->find($archetypeIdOrCode);
+        } else {
+            $archetype = $this->get('sylius.repository.product_archetype')->findOneByCode($archetypeIdOrCode);
+        }
```

Comment:
We could extract this whole ``if`` to separate method and then:
- check if given ``archetype`` attribute is int (maybe ``is_int`` function would be better solution?)
- early return archetype get by ID
- if parameter is string, return archetype get by code

It would improve code readability and allows to get rid of nasty ``else`` statement ;)